src/main/java/org/broadinstitute/hellbender/utils/io/IOUtils.java

```diff
@@ -564,14 +565,36 @@ public static void assertFileIsReadable(final File file) {
     public static void assertFileIsReadable(final Path path) {
         Utils.nonNull(path);
 
-        if ( ! Files.exists(path) ) {
-            throw new UserException.CouldNotReadInputFile(path, "It doesn't exist.");
-        }
-        if ( ! Files.isRegularFile(path) ) {
-            throw new UserException.CouldNotReadInputFile(path, "It isn't a regular file");
-        }
-        if ( ! Files.isReadable(path) ) {
-            throw new UserException.CouldNotReadInputFile(path, "It is not readable, check the file permissions");
+        try {
+            if ( ! Files.exists(path) ) {
+                throw new UserException.CouldNotReadInputFile(path, "It doesn't exist.");
+            }
+            if ( ! Files.isRegularFile(path) ) {
+                throw new UserException.CouldNotReadInputFile(path, "It isn't a regular file");
+            }
+            if ( ! Files.isReadable(path) ) {
+                throw new UserException.CouldNotReadInputFile(path, "It is not readable, check the file permissions");
+            }
+        } catch (com.google.cloud.storage.StorageException cloudBoom) {
+            // probably a permissions problem, or perhaps a disabled account.
+            logger.debug(cloudBoom.toString());
+            logger.debug("code: " + cloudBoom.getCode());
+            logger.debug("location: " + cloudBoom.getLocation());
+            logger.debug("reason: " + cloudBoom.getReason());
```

Comment:
[Here is the documentation](https://cloud.google.com/storage/docs/json_api/v1/status-codes). 
`reason` is a short code like "invalidParameter" and `message` is meant for human consumption. `location` is the name of the parameter that caused the error. `path` of course is the argument to the method, the file we're trying to make sure is readable.